Build for master broken on OS X

Crap, missed something here in one of the ``ndimage`` refactoring PRs - the build is broken on OS X as of commit 9b576070d (Python 3.5, NumPy 1.13.0, OS X 10.12, Clang 8.1.0):

        scipy/ndimage/src/ni_morphology.c:470:13: warning: comparison of integers of different signs: 'npy_uint' (aka 'unsigned int') and 'int' [-Wsign-compare]
                    CASE_ERODE_POINT2(NPY_UINT, npy_uint,
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        scipy/ndimage/src/ni_morphology.c:317:55: note: expanded from macro 'CASE_ERODE_POINT2'
                if (_to != _bf_value && *(_type *)(_pi + _to) == _true) {             \
                                        ~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~
        scipy/ndimage/src/ni_morphology.c:476:13: warning: comparison of integers of different signs: 'npy_ulong' (aka 'unsigned long') and 'int' [-Wsign-compare]
                    CASE_ERODE_POINT2(NPY_ULONG, npy_ulong,
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        scipy/ndimage/src/ni_morphology.c:317:55: note: expanded from macro 'CASE_ERODE_POINT2'
                if (_to != _bf_value && *(_type *)(_pi + _to) == _true) {             \
                                        ~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~
        scipy/ndimage/src/ni_morphology.c:482:13: warning: comparison of integers of different signs: 'npy_ulonglong' (aka 'unsigned long long') and 'int' [-Wsign-compare]
                    CASE_ERODE_POINT2(NPY_ULONGLONG, npy_ulonglong,
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        scipy/ndimage/src/ni_morphology.c:317:55: note: expanded from macro 'CASE_ERODE_POINT2'
                if (_to != _bf_value && *(_type *)(_pi + _to) == _true) {             \
                                        ~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~
        In file included from scipy/ndimage/src/ni_morphology.c:32:
        In file included from scipy/ndimage/src/ni_support.h:42:
        In file included from scipy/ndimage/src/nd_image.h:38:
        In file included from /Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/numpy/core/include/numpy/arrayobject.h:4:
        In file included from /Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/numpy/core/include/numpy/ndarrayobject.h:27:
        /Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/numpy/core/include/numpy/__multiarray_api.h:1453:1: warning: unused function '_import_array' [-Wunused-function]
        _import_array(void)
        ^
        4 warnings generated.
        /usr/bin/clang -bundle -undefined dynamic_lookup -arch i386 -arch x86_64 -g build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_filters.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_fourier.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_interpolation.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_measure.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_morphology.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_support.o -Lbuild/temp.macosx-10.6-intel-3.5 -o scipy/ndimage/_nd_image.cpython-35m-darwin.so
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_filters.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_fourier.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_interpolation.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_measure.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_support.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_morphology.o
        ld: 6 duplicate symbols for architecture i386
        clang: error: linker command failed with exit code 1 (use -v to see invocation)
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_filters.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_fourier.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_interpolation.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_measure.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_support.o
        duplicate symbol __scipy_ndimage_ARRAY_API in:
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o
            build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_morphology.o
        ld: 6 duplicate symbols for architecture i386
        clang: error: linker command failed with exit code 1 (use -v to see invocation)
        error: Command "/usr/bin/clang -bundle -undefined dynamic_lookup -arch i386 -arch x86_64 -g build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/nd_image.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_filters.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_fourier.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_interpolation.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_measure.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_morphology.o build/temp.macosx-10.6-intel-3.5/scipy/ndimage/src/ni_support.o -Lbuild/temp.macosx-10.6-intel-3.5 -o scipy/ndimage/_nd_image.cpython-35m-darwin.so" failed with exit status 1

Will investigate tomorrow. Works fine on Linux, not clear what the difference is.

Also some warnings about unsigned int comparisons (more of the same that I didn't copy in).

I cannot reproduce the failure, but my build seems to only be doing the `-arch x86_64`.

There is something fishy, though... Here's a warning I'm seeing:

```
creating build/temp.macosx-10.5-x86_64-3.5/scipy/ndimage
creating build/temp.macosx-10.5-x86_64-3.5/scipy/ndimage/src
compile options: '-DNPY_NO_DEPRECATED_API=NPY_1_9_API_VERSION -DNO_ATLAS_INFO=3 -DHAVE_CBLAS -Iscipy/ndimage/src -I/Users/jaimefrio/miniconda/envs/numpydev35/lib/python3.5/site-packages/numpy-1.14.0.dev0+7a85008-py3.5-macosx-10.5-x86_64.egg/numpy/core/include -Iscipy/_lib/src -I/Users/jaimefrio/miniconda/envs/numpydev35/lib/python3.5/site-packages/numpy-1.14.0.dev0+7a85008-py3.5-macosx-10.5-x86_64.egg/numpy/core/include -I/Users/jaimefrio/miniconda/envs/numpydev35/include/python3.5m -c'
gcc: scipy/ndimage/src/ni_interpolation.c
In file included from scipy/ndimage/src/ni_interpolation.c:32:
In file included from scipy/ndimage/src/ni_support.h:42:
In file included from scipy/ndimage/src/nd_image.h:38:
In file included from /Users/jaimefrio/miniconda/envs/numpydev35/lib/python3.5/site-packages/numpy-1.14.0.dev0+7a85008-py3.5-macosx-10.5-x86_64.egg/numpy/core/include/numpy/arrayobject.h:4:
In file included from /Users/jaimefrio/miniconda/envs/numpydev35/lib/python3.5/site-packages/numpy-1.14.0.dev0+7a85008-py3.5-macosx-10.5-x86_64.egg/numpy/core/include/numpy/ndarrayobject.h:27:
/Users/jaimefrio/miniconda/envs/numpydev35/lib/python3.5/site-packages/numpy-1.14.0.dev0+7a85008-py3.5-macosx-10.5-x86_64.egg/numpy/core/include/numpy/__multiarray_api.h:1453:1: warning: unused function '_import_array' [-Wunused-function]
_import_array(void)
^
1 warning generated.
```

Here is some code created by [`generate_numpy_api.py`](https://github.com/numpy/numpy/blob/master/numpy/core/code_generators/generate_numpy_api.py):

```
#if defined(PY_ARRAY_UNIQUE_SYMBOL)
#define PyArray_API PY_ARRAY_UNIQUE_SYMBOL
#endif
#if defined(NO_IMPORT) || defined(NO_IMPORT_ARRAY)
extern void **PyArray_API;
#else
#if defined(PY_ARRAY_UNIQUE_SYMBOL)
void **PyArray_API;
#else
static void **PyArray_API=NULL;
#endif
#endif

...

#if !defined(NO_IMPORT_ARRAY) && !defined(NO_IMPORT)
static int
_import_array(void)
{
...
```

In theory things work as expected because we have `#define NO_IMPORT_ARRAY` and thus the numpy API is declared as `extern void **PyArray_API;`. But if `NO_IMPORT_ARRAY` is defined, then `_import_array` should not be defined, and we shouldn't be getting that warning.

--- a/scipy/ndimage/src/ni_support.h
+++ b/scipy/ndimage/src/ni_support.h
@@ -38,9 +38,9 @@
  * unit that initializes the shared numpy API by calling import_array()
  * must bypass this by explicitly including nd_image.h before ni_support.h.
  */
-#define NO_ARRAY_IMPORT
+#define NO_IMPORT_ARRAY
 #include "nd_image.h"
-#undef NO_ARRAY_IMPORT
+#undef NO_IMPORT_ARRAY
 
 #include <stdlib.h>
 #include <float.h>

